generator/internal/scaffold_generator.cc

```diff
@@ -149,11 +214,77 @@ void MakeDirectory(std::string const& path) {
 #endif  // _WIN32
 }
 
+void GenerateMetadata(
+    std::map<std::string, std::string> const& vars,
+    std::string const& output_path,
+    google::cloud::cpp::generator::ServiceConfiguration const& service,
+    bool allow_placeholders) {
+  auto const destination = [&]() {
+    MakeDirectory(output_path);
+    auto d = output_path + "/" + LibraryPath(service.product_path());
+    MakeDirectory(d);
+    auto l = vars.find("service_subdirectory");
+    if (l == vars.end()) return d;
+    if (l->second.empty()) return d;
+    d += l->second;
```

Comment:
will the subsequent `MakeDirectory(d)` call work if `l->second` is like `foo/v1` ?

Reply:
Don't know, but that is what the rest of the scaffold does.